src/modules/aws_iam@0.0.1/entity/role.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  Column,
+  Entity,
+  JoinColumn,
+  ManyToOne,
+  OneToMany,
+  PrimaryColumn,
+} from 'typeorm';
+
+import { cloudId, } from '../../../services/cloud-id'
+
+// TODO complete schema
+@Entity()
+export class Role {
+  @Column({
+    nullable: true,
+  })
+  @cloudId
+  arn?: string;
+
+  // Guaranteed unique in AWS
+  // Maximum 128 characters. Use alphanumeric and '+=,.@-_' characters.
+  @PrimaryColumn()
+  roleName: string;
+
+  @Column()
+  assumeRolePolicyDocument: string;
+
+  @Column({
+    nullable: true,
+  })
+  description?: string;
+
+  @Column({
+    type: 'text',
+    array: true,
+    default: [],
+  })
+  attachedPoliciesArns: string[];
+
+  @OneToMany(() => InlineRolePolicy, irp => irp.role, {
+    eager: true,
+    cascade: true
+  })
+  inlineRolePolicies: InlineRolePolicy[];
+}
+
+@Entity()
+export class InlineRolePolicy {
+  @PrimaryColumn()
+  policyName: string;
+
+  @Column()
+  policy: string;
+
+  @ManyToOne(() => Role)
+  @JoinColumn({
+    name: 'role_name',
+  })
+  role: Role;
+}
```

Comment:
I still think we are not being consistent with other modules not having the auto incremental `id` column in these entities

Reply:
the role name is guaranteed unique in aws, so it simplifies references to the table and reduces aws calls since all their methods just take the role name. in some cases we need the id, but we shouldn't have it if adds complexity? there are a few modules where we could remove it too

Reply:
Because it's set by the user instead of set by AWS, I think it's fine? We did the incrementing IDs on the others because the unique ID for the records was controlled by AWS for the other modules.